crates/project/src/project.rs

```diff
@@ -11730,6 +11769,7 @@ async fn load_shell_environment(dir: &Path) -> Result<HashMap<String, String>> {
 
     let output = smol::process::Command::new(&shell)
         .args(["-i", "-c", &command])
+        .envs(direnv_environment)
```

Comment:
Why do we have to spawn a login shell with that environment? Wouldn't the better approach be to collect both environments and then merge them?

Reply:
The shell does it for us, so I'm basically leveraging it's infrastructure to help us not do additional work. It also has some other upsides, imagine if someone uses direnv to install some other local environment tool, well, now we support that, very specific usecase too

Reply:
Yeah, it does, but is it correct? Technically you only get that `direnv` environment once you're inside the directory. Now we set it before you get to that directory and then `cd` into it. Doesn't that have the potential to mess up some other shell hooks?

Reply:
If you have multiple shell hooks, I don't think their order is really determined anyways, so I think this is just equivalent to direnv executing first. As far as I know, it won't mess itself up if you already have the environment set before the `cd`, and anything else would've probably executed after it in some case anyways. The shell behaviours are sometimes strange though, so I can't guarantee that this is correct. Can we maybe try doing it and waiting for any potential bug reports, or adding this under an opt-in setting?

Reply:
I've added a setting for that, defaulting to the old behaviour. I think that should resolve this issue